### PR TITLE
feat: add "ensure seccomp not unconfined" rule

### PR DESCRIPTION
**Pull request by Heydad-Helfer**
Added the new schema, used defs for reusability. 
Documentation link is still needed.

### CODE REVIEW

**Comment by eyarz** [comment]
@Heydad-Helfer Thank you for the PR!
Do you also plan to write the docs for this rule?
https://github.com/datreeio/docs

**Comment by Heydad-Helfer** [comment]
> @Heydad-Helfer Thank you for the PR! Do you also plan to write the docs for this rule? https://github.com/datreeio/docs

Hi. I can add this, but it'll take me some time. Do you want to wait on the merge until a documentation is available?

**Comment by eyarz** [comment]
NP, we can wait :)
it will take us a few days to review your PR anyway...

File: pkg/defaultRules/defaultRules.yaml
Comment (eyarz) [inline pkg/defaultRules/defaultRules.yaml]: I think it will be more clear to the user if you will use "prevent" instead of "ensure ... not".
so maybe "Prevent setting 'Unconfined' as seccomp profile"?

File: pkg/defaultRules/defaultRules.yaml
Comment (eyarz) [inline pkg/defaultRules/defaultRules.yaml:1879]: from the K8s docs:
`...you can configure Kubernetes to use the profile that the container runtime prefers by default, rather than falling back to Unconfined`

so, basically, not setting `seccompProfile` means you are using `Unconfined` as your seccomp profile.
also, `seccompProfile` supports only the values "Localhost", "RuntimeDefault", "Unconfined", so maybe a validation that "Localhost" or "RuntimeDefault" value is set will be more comprehensive.
WDYT?

File: pkg/defaultRules/defaultRules.yaml
Comment (eyarz) [inline pkg/defaultRules/defaultRules.yaml]: let's set it to false because it's too aggressive rule to enforce on all users

**Review by eyarz — CHANGES_REQUESTED** [review]
@Heydad-Helfer please review my comments :)

File: pkg/defaultRules/defaultRules.yaml
Comment (Heydad-Helfer) [reply pkg/defaultRules/defaultRules.yaml:1879]: So should check be that the seccompProfile *does* defined in the config, and that it's only "Localhost" or "RuntimeDefault"?

File: pkg/defaultRules/defaultRules.yaml
Comment (Heydad-Helfer) [reply pkg/defaultRules/defaultRules.yaml]: Modified the message

File: pkg/defaultRules/defaultRules.yaml
Comment (Heydad-Helfer) [reply pkg/defaultRules/defaultRules.yaml]: Set the enableOnDefault to false

File: pkg/defaultRules/defaultRules.yaml
Comment (eyarz) [reply pkg/defaultRules/defaultRules.yaml:1879]: I think this should be the logic:
seccompProfile has to be set && it has to be "Localhost" or "RuntimeDefault

Comment (Heydad-Helfer) [reply pkg/defaultRules/defaultRules.yaml:1879]: OK, so the policy should check that seccompProfile is either defined globally and container overrides are optional, or is must be defined for each and every container. And in any case, it must not be "Unconfined".

I'll start working on it.

**Comment by eyarz** [comment]
@Heydad-Helfer is this ready for code review?

**Comment by Heydad-Helfer** [comment]
Hi, unfortunately not yet, as I didn't have much time to finish this. I'll
work on it this weekend.

File: pkg/defaultRules/defaultRules.yaml
Comment (Heydad-Helfer) [reply pkg/defaultRules/defaultRules.yaml:1879]: @eyarz please check the policy

**Comment by eyarz** [comment]
@Heydad-Helfer no worries, take your time.
just ping me when I should review it again :)

File: pkg/defaultRules/defaultRules.yaml
Comment (hadar-co) [inline pkg/defaultRules/defaultRules.yaml]: The name should be about what this rule does, and not how it does it. Something like:
"Prevent containers from having unnecessary system call privileges"
what do you think?

File: pkg/defaultRules/defaultRules.yaml
Comment (hadar-co) [inline pkg/defaultRules/defaultRules.yaml]: Also, to keep a consistent format of our rule docs, I think this should be:
"Incorrect value for key `seccompProfile` - set an explicit value to prevent malicious use of system calls within the container"

**Comment by hadar-co** [comment]
@Heydad-Helfer Hey Heydad, left you some comments on your PR:)

Also, we created the docs for this rule:
https://github.com/datreeio/docs/pull/89

**Comment by Heydad-Helfer** [comment]
Hi. Thanks.
I'll make the changes.

File: pkg/defaultRules/defaultRules.yaml
Comment (hadar-co) [inline pkg/defaultRules/defaultRules.yaml:1934]: Add the following rows (otherwise this will not work for Deployments):
additionalProperties:
  $ref: "#"
items:
  $ref: "#"

**Review by hadar-co — CHANGES_REQUESTED** [review]

**Comment by hadar-co** [comment]
@Heydad-Helfer Thanks, just a couple of last things - 
1. I left a comment regarding some lines to add to support Deployments as well (currently the rule will only work for Pods).
2. All of our rules have test files under pkg/policy/tests (1 passing manifest and 1 failing for each rule). Can you please add test files for your rule? They should be called 61-pass.yaml and 61-fail.yaml. 

And that should be it! 🥳

**Comment by Heydad-Helfer** [comment]
Hi @hadar-co , I pushed the fixes. Please review.
Thanks!

**Review by hadar-co — APPROVED** [review]

**Review by eyarz — APPROVED** [review]

**Comment by hadar-co** [comment]
@Heydad-Helfer Awesome:) thanks for the PR 🤘🏽

**Pull request merged**
